### ISSUE TITLE
After recent High Sierra update, cannot access accessibility features

### ISSUE THREAD

**Issue by garrettjj** [koekeishiya/chunkwm#295]
After a recent update to High Sierra, chunkwm stopped tiling. Tried manually starting is via Terminal.app, receiving warning that "chunkwm: could not access accessibility features! abort.." When enabling service, it repeatedly asks for Accessibility access. I've verified it does have it set.

Here's the short list of attempts at fixing:
- Toggled Accessibility off/on
- Reinstalling via Homebrew
- Reinstalling from GitHub branch via Homebrew
- Restart Finder.app

Any advice for other things to try would be fantastic.

**Comment by koekeishiya** [comment]
You need to remove chunkwm from the accessibliy list.
Start chunkwm and accept the prompt, add it to accessiblity list.
Start chunkwm again.

If you use brew, this amounts to the following steps:

`brew services stop chunkwm`
`open accessiblity list and remove chunkwm`
`brew services start chunkwm`
`accept promt, open accessiblity and check chunkwm`
`brew services stop chunkwm`
`brew services start chunkwm`

The reason for this is due to how the launchd service works.

**Comment by garrettjj** [comment]
Thanks for the quick response and the great piece of software!

I went through these steps and I'm still receiving the same error message. I also repeated them while calling chunkwm manually so I could try cutting homebrew out for the time being. 

If there's any information or steps to take that would be helpful in identifying the issue, please let me know.

**Comment by koekeishiya** [comment]
I'm not really sure what the issue could be here to be honest.
The best suggestion I can give  would be to do a full removal, reboot the system and do an install.
Before starting the brew service, run chunkwm manually from a terminal to grant it accessibility access, and then run it through the brew service.

**Comment by danielzfranklin** [comment]
Hi,
I wasn't sure if I should create a new issue. I've tried the same suggestions, and also have had no success. When I start chunkwm through the terminal, I get prompted to grant accessibility access to the terminal, not to chunkwm. When I start chunkwm through brew, I get prompted to grant accessibility access to chunkwm.

With the following lines at the top of my ~/.chunkwmrc, I see no log messages.
```bash
chunkc core::log_level info
chunkc core::log_file ~/chunkwm.log
```

If I grant accessibility access to my terminal, and stop the brew chunkwm services, I see the following after running chunkwm in a terminal.
```
2018-02-17 13:51:18.914 chunkwm[3007:21941] ApplePersistence=NO
```

**Comment by Wes974** [comment]
Try that : 
Remove chunkwm in the accessibility panel and then remove chunkwm with `brew remove --force chunkwm`. 

Then open Keychain Access and create a certificate from ` Keychain Access > Certificate Assistant > Create a certificate `.
In there choose the name `chunkwm-cert` and set the type to be code signing, like so :

<img width="628" alt="image" src="https://user-images.githubusercontent.com/7169107/38416656-ba0afeb0-3995-11e8-85db-ab4eaa448648.png">

After you have create it, go to the terminal and then install chuckwm with `brew install chunkwm`.
Once it is installed launch this command `codesign -fs "chunkwm-cert" /usr/local/opt/chunkwm/bin/chunkwm` it will sign the chunkwm binary with the code signing identity we created earlier.

Once it is done launch chunkwm with `brew services start chunkwm`. It will still request for you to enable the accessibility service. Enables it, and then restart chunkwm with `brew services restart chunkwm`.

And then it should be good. After each update of chunkwm before launching it you will have to resign the binary with `codesign -fs "chunkwm-cert" /usr/local/opt/chunkwm/bin/chunkwm` but it wont prompt you to enable the accessibility services anymore.

**Comment by koekeishiya** [comment]
There should be enough information here to resolve this problem. Closing.
